bodhi/tests/functional/test_updates.py

```diff
@@ -2520,6 +2520,69 @@ def test_edit_testing_update_reset_karma_with_same_tester(self, publish, *args):
 
     @mock.patch(**mock_valid_requirements)
     @mock.patch('bodhi.notifications.publish')
+    def test_for_negative_karma(self, publish, *args):
+        """
+        Checks if an update has any negative karma
+        The test asserts True for has_negative_karma i.e, if there exists a negative karma
+        The test asserts False for has_negative_karma i.e, if there exists no negative karma
+        """
+        user = User(name=u'bob')
+        self.db.add(user)
+        self.db.flush()
+
+        nvr = 'bodhi-2.1-1.fc17'
+        args = self.get_update(nvr)
+        resp = self.app.post_json('/updates/', args).json_body
+        publish.assert_called_with(topic='update.request.testing', msg=ANY)
+
+        up = self.db.query(Update).filter_by(title=nvr).one()
+        up.request = None
+        up.status = UpdateStatus.testing
+        self.db.flush()
+
+        up.comment(self.db, u'LGTM', author=u'mac', karma=1)
+        up = self.db.query(Update).filter_by(title=nvr).one()
+        self.assertFalse(up.has_negative_karma)
+
+        up.comment(self.db, u'Failed to work', author=u'ralph', karma=-1)
+        up = self.db.query(Update).filter_by(title=nvr).one()
+        self.assertTrue(up.has_negative_karma)
+
+        up.comment(self.db, u'Doesn\'t work', author=u'bowlofeggs', karma=-1)
+        up = self.db.query(Update).filter_by(title=nvr).one()
+        self.assertTrue(up.has_negative_karma)
```

Comment:
As I mentioned, please split this method up into several tests. There are some more tests you should add too, for example, a test where a user sets negative karma in one comment, and then positive karma in a later comment. You should also run up.has_negative_karma in a test where there are no comments to make sure it returns False.